src/js/utils.js

```diff
@@ -322,6 +322,35 @@ function parseCookie(str, opts) {
   return parsed;
 }
 
+function CheckedDNTBuffer(size) {
+  if (size) {
+    this.size = size;
+  }
+  this.index = 0;
+  this.array = new Array(this.size);
+  this.set = new Set();
+}
+
+CheckedDNTBuffer.prototype = {
+  size: 1000,
+  // if we've seen item, return true. if we haven't store it and return false
+  hasOrSet: function(item) {
```

Comment:
I agree `hasOrSet` is a bad name, I'm open to suggestions.

 But I don't think adding two functions is the correct thing to do, I actually considered mimicing the `Set` api first (like you suggest). However if `add` gets put into the callback, then the domain won't get added until the dnt check finishes, which could take a while, possibly after we ask for another DNT check for the same domain.

I also think having a seperate `add` and `has` could lead to misuse of the API. For example `add` might be called on an `item` that is already in the buffer. I could guard against this inside add to make it it idempotent, but then I'd be checking the set for containment of the domain twice for my main use case which is what happenes in the `hasOrSet` function.